app/models/nonprofit.rb

```diff
@@ -180,6 +181,10 @@ def as_json(options = {})
     h
   end
 
+  def state_in_us?
+    errors.add(:state_not_in_us, 'Only US states are allowed') unless ISO3166::Country[:US].subdivisions[state_code]
```

Comment:
When you add the errors on validation, you're often validating the attributes which have been passed in. In this case, you're validating `state_code`. The way to indicate that in ActiveRecord errors is to pass the attribute as a symbol to the first parameter, i.e. pass `:state_code`.

Additionally, as you noticed in your test, Rails displays errors as a phrase. Therefore, the second parameter needs to be formatted as the rest of the phrase.

A good phrase for this might be "State code must be a US two-letter state code".


One last thing: I think you can use `has_key?` to check the `state_code` just like I mention in the comment down in `config/state_code_constraint.rb`. That would be a better choice.

Reply:
We agree on it, much appreciated!